SU-HSDO/suhumsci
# Feat(STN-810): callout box well

# READY FOR REVIEW

## Summary
Add a well option for the Callout Box paragraph component.

- This work adds a "Well" view mode that is being used by Callout Box display modes.
- There is a new field added using the View Mode Select field to allow you to choose between default or Well.
- Default is the standard look, so no hooks for overriding is required.
- This work adds a template for when a Well option to Callout Box is added.

JIRA Card: https://sparkbox.atlassian.net/browse/STN-810

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. npm run test
2. Import the config `lando drush @sparkbox_sandbox.local cim --partial`
2. clear cache `lando drush @sparkbox_sandbox.local cr`
3. Create a flexible page using Callout Box paragraph components.
4. Add a Well and a default option for testing the use.

- [x] Verify the Callout Box Well and Default both still float as they should when above a text area.
- [x] Verify both left and right floats
- [x] Verify the Well Callout box does not float above any of the other paragraph component types.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
